### PR TITLE
Added bulk insert/update feature

### PR DESCRIPTION
**Pull request by corcoja**
## Description

Added a new method to database to modify multiple documents with a single request.

## Motivation and Context

I need to remove or update multiple documents at once by providing a list of JSON objects containing `_id` and `_rev`.

## How Has This Been Tested?

I tested this feature on my machine with macOS Sierra and CouchDB installed as well as on Bluemix platform with an app using IBM Cloudant service.

## Checklist:

- [x] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [x] If applicable, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by na-gupta** [comment]
@iuliancorcoja Thank you for this PR. It looks good to me. Could you please add a unit test for the new function?

**Comment by na-gupta** [comment]
The travis build will fail because of an issue accessing credentials by users outside the org. So don't be concerned about the build failing.

**Comment by corcoja** [comment]
@na-gupta, I've added 3 unit tests for bulk insert, update and delete. Please let me know if there is anything else needed.

Regards,
Iulian.

File: Tests/CouchDBTests/DocumentBulkUpdateTests.swift [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	// To enable running Linux and OSX tests in parallel
+#if os(Linux)
+	let dbName = "test_db_linux"
+#else
+	let dbName = "test_db_db"
+#endif
+
+	// MARK: - Database connection properties
+
+	static let dbSecuredConnection = true
```
Comment (na-gupta) [inline Tests/CouchDBTests/DocumentBulkUpdateTests.swift]: Please set `dbSecuredConnection` to false. Our testing cloudant instance is not setup for https connections.

**Review by na-gupta — CHANGES_REQUESTED** [review]
@iuliancorcoja  Looks great. Just a couple of small changes needed to the tests:
- Please add DocumentBulkUpdateTests to Tests/LinuxMain.swift so they run on Linux builds as well.
- Please set dbSecuredConnection to false. Our testing cloudant instance is not setup for https connections.

**Comment by corcoja** [comment]
@na-gupta, done!

**Review by na-gupta — APPROVED** [review]

**Comment by na-gupta** [comment]
@iuliancorcoja Merged and tagged (1.7.1). Thank you!

**Pull request merged**